epilepsy12/tasks.py

```diff
@@ -69,10 +73,288 @@ def async_insert_old_pt_data(csv_path: str = "data.csv"):
+                    final_list.append(item)  
+        
+        # Group organisation body by KPI, then add to dataframe - collect all values relating to KPI 1 across all organisation bodies, add to dataframe, repeat for next KPI
+
+        else:
+            for kpi in measures:
+                if is_regional:
+                    if wales_region_object[f"{kpi}_total_eligible"] == 0:
+                        item = {
+                        title: "Health Boards",
+                        "Measure": kpi,
+                        "Percentage": 0,
+                        "Numerator": wales_region_object[f"{kpi}_passed"],
+                        "Denominator": wales_region_object[f"{kpi}_total_eligible"],
+                        }
+                    else:
+                        item = {
+                            title: "Health Boards",
+                            "Measure": kpi,
+                            "Percentage": wales_region_object[f"{kpi}_passed"]
+                            / wales_region_object[f"{kpi}_total_eligible"]
+                            * 100,
+                            "Numerator": wales_region_object[f"{kpi}_passed"],
+                            "Denominator": wales_region_object[f"{kpi}_total_eligible"],
+                        }
+                    final_list.append(item)
+                for key in objects:
+                    object = objects[key]
+                    if object == None:
+                        item = {
+                            title: key,
+                            "Measure": kpi,
+                            "Percentage": 0,
+                            "Numerator": 0,
+                            "Denominator": 0,
+                        }
+                    elif object[f"{kpi}_total_eligible"] == 0:
+                        item = {
+                            title: key,
+                            "Measure": kpi,
+                            "Percentage": 0,
+                            "Numerator": object[f"{kpi}_passed"],
+                            "Denominator": object[f"{kpi}_total_eligible"],
+                        }
+                    else:
+                        item = {
+                            title: key,
+                            "Measure": kpi,
+                            "Percentage": object[f"{kpi}_passed"]
+                            / object[f"{kpi}_total_eligible"]
+                            * 100,
+                            "Numerator": object[f"{kpi}_passed"],
+                            "Denominator": object[f"{kpi}_total_eligible"],
+                        }
+                    final_list.append(item) 
+        
+        return pd.DataFrame.from_dict(final_list)
+
+    # COUNTRY - SHEET 1
+    # create a dataframe with a row for each measure of each country, and a column for each of ["Measure", "Percentage", "Numerator", "Denominator"]
+    
+    CountryKPIAggregation = apps.get_model("epilepsy12", "CountryKPIAggregation")
```

Comment:
That's exactly it - we only calculate England and Wales (not Scotland or NI). Also England and Wales need to be in the same dataframe. Happy to put this into a function though if you think it would be tidier?

Reply:
National covers both England and Wales.

Reply:
Yep National does indeed. Sheet 1 specifies per Country, so we have to split it up into England and Wales and use the CountryKPIAggregation.